core/main/src/processor/main_context_processor.rs

```diff
@@ -233,9 +214,18 @@ impl ExtnEventProcessor for MainContextProcessor {
         if let Some(update) = &extracted_message.update_type {
             match update {
                 RippleContextUpdateType::TokenChanged => {
-                    if let ActivationStatus::AccountToken(_t) = &extracted_message.activation_status
+                    if let ActivationStatus::AccountToken(t) = &extracted_message.activation_status
                     {
-                        Self::initialize_token(&state.state).await
+                        //Call initialize token only when account session was not initialized the first time
+                        if state.state.session_state.get_account_session().is_none() {
+                            Self::initialize_token(&state.state).await
```

Comment:
Aren't we still calling `initialize_token` on startup, so this will potentially get called twice? There's stuff in there that is truly initialization i.e. do once.

Reply:
Should `initialize_token` be refactored to extract the `check_account_session_token` call?

Reply:
yes we addressed this in the PR. 

1. During bootstrap if account_session is available we will just call ripple context update
2. During Token update we will first check if the account_session is already available if so we will just update token.
If account token is not available we will call initialize_token to make sure we kick start the process for token interactions